src/utils/mappingUtil.ts

```diff
@@ -0,0 +1,104 @@
+import { ValidationResult } from "joi";
+import { isArray } from "lodash";
+
+import * as enums from "../assets/Enums";
+import * as models from "../models";
+import * as validators from "../models/validators";
+
+export class MappingUtil {
+
+  public static manageDefectsArray(testResult: models.ITestResult) {
+    testResult.testTypes.forEach((testType: models.TestType) => {
+      if (enums.SPECIALIST_TEST_TYPE_IDS.includes(testType.testTypeId)) {
+        testType.defects = [];
+      }
+    });
+  }
+
+    public static setCreatedAtAndLastUpdatedAtDates(payload: models.ITestResultPayload): models.ITestResultPayload {
+        const createdAtDate = new Date().toISOString();
+        payload.createdAt = createdAtDate;
+        payload.createdById = payload.testerStaffId;
+        payload.createdByName = payload.testerName;
+        payload.testVersion = enums.TEST_VERSION.CURRENT;
+        payload.reasonForCreation = enums.REASON_FOR_CREATION.TEST_CONDUCTED;
+        if (payload.testTypes.length > 0) {
```

Comment:
This will probably be refactor later but I think we have to try to use guard close as much as possible so we avoid nested if and exist early if the condition is not met

```
if (!payload.testTypes.length > 0) return
// else do stuff, process data to merge payload
```

Reply:
will be done in 18782